_posts/en/newsletters/2023-12-06-newsletter.md

```diff
@@ -0,0 +1,177 @@
+---
+title: 'Bitcoin Optech Newsletter #280'
+permalink: /en/newsletters/2023/12/06/
+name: 2023-12-06-newsletter
+slug: 2023-12-06-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter FIXME
+
+## News
+
+- **Cluster mempool discussion:** Bitcoin Core developers working on
+  [cluster mempool][topic cluster mempool] started a [working
+  group][wg-cluster-mempool] (WG) on the Delving Bitcoin forum. Cluster
+  mempool is a proposal to make it much easier to compare the
+  profitability of different valid orderings of transactions.  A valid
+  ordering of Bitcoin transactions requires parent transactions get
+  confirmed before their children, either by putting the parent in an
+  earlier block than any of its children, or earlier in the same block.
+  _Clusters_ of one or more related transactions are designed to be easy
+  to include into feerate-sorted _chunks_ that are each independent of
+  the other chunks. For example, we might split a mempool into 100
+  chunks each about 4 million weight in size, each of which could be
```

Comment:
Having calculated a linearization for every cluster in our mempool and subdivided it into chunks, we can now create a total order of all chunks in our mempool. We only need to compare the best unselected chunks in each cluster, rather than each transaction with its ancestry. By picking the best unselected chunk from a cluster repeatedly, we can easily assemble a block template, and vice versa, if we evict, we can simply compare the worst feerate chunks from each cluster to see which one we would pick into a block last.

Reply:
@murchandamus Note that to assemble the block we're pick the best chunks from *all* clusters in the mempool, simply comparing them by feerate (and perhaps size).

@sipa It makes sense to me that chunks need to be limited in size (as we add chunks to a block as individual units), but not why clusters necessarily need to have the same limit. I think the cluster limit results from the possibility that the optimal chunking of a cluster may end up being almost the entire cluster (particularly when multiple children pay for a parent, eg), and if we didn't limit the cluster size, we could end up stuck with a suboptimal linearisation, which we don't want. Maybe some commentary about limits would make for a good post as well?